sky/packages/sky/lib/widgets/transitions.dart

```diff
@@ -206,3 +206,38 @@ class SquashTransition extends TransitionBase {
     return new SizedBox(width: _maybe(width), height: _maybe(height), child: child);
   }
 }
+
+class BuilderTransition extends TransitionBase {
+  BuilderTransition({
+    Key key,
+    this.variables,
+    this.builder,
+    Duration duration,
+    AnimationPerformance performance,
+    Direction direction,
+    Function onDismissed,
+    Function onCompleted,
+    Widget child
+  }) : super(key: key,
+             duration: duration,
+             performance: performance,
+             direction: direction,
+             onDismissed: onDismissed,
+             onCompleted: onCompleted,
+             child: child);
+
+  List<AnimatedValue> variables;
+  Function builder;
+
+  void syncFields(BuilderTransition updated) {
+    variables = updated.variables;
+    builder = updated.builder;
+    super.syncFields(updated);
+  }
+
+  Widget build() {
+    for (int i = 0; i < variables.length; ++i)
+      performance.updateVariable(variables[i]);
+    return builder(variables);
```

Comment:
Maybe instead of handing back the list of variables, the code with the builder function should be responsible for holding on to them.  That way they can have reasonable names instead of variable[0]